### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by JaSei**

### CODE REVIEW

File: append.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package pathutil
+
+import (
+	"github.com/pkg/errors"
+)
+
+func (path PathImpl) Append(data string) error {
+	return path.AppendBytes([]byte(data))
```
Comment (ondrej-hosak) [inline append.go:8]: You don't have to handle open error here?

File: append_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package pathutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAppend(t *testing.T) {
```
Comment (ondrej-hosak) [inline append_test.go:9]: I don't see tests for AppendBytes method.

File: makepath_test.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package pathutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMakePath(t *testing.T) {
```
Comment (ondrej-hosak) [inline makepath_test.go:9]: I dont't see tests for MakePathMode method

**Review by ondrej-hosak — APPROVED** [review]
This PR is very large, so it's hard to go trough everything.

File: append.go (hunk 1 above)
Comment (JaSei) [reply append.go:8]: no isn't open handle, only return error

File: append_test.go (hunk 2 above)
Comment (JaSei) [reply append_test.go]: good point, I do it...

File: makepath_test.go (hunk 3 above)
Comment (JaSei) [reply makepath_test.go]: Good point, I do it...

File: append_test.go (hunk 2 above)
Comment (JaSei) [reply append_test.go:9]: Because `Append` use internal `AppendBytes`

File: makepath_test.go (hunk 3 above)
Comment (JaSei) [reply makepath_test.go:9]: becuase `MakePath` use internal `MakePathMode`

**Pull request merged**